### DISCUSSION TITLE
Why is dosbox staging video output so pixelated compared to other versions of Dosbox?

### DISCUSSION THREAD

**Discussion by Paolo86cripple** [dosbox-staging/dosbox-staging#2157] [Q&A] [answered]
Hello. I have come across a strange issue compared to other versions Dosbox I was using. The output, both in CLI and in-game, is pixelated. Already  tried  specifying opengl render mode in the conf file. Here are a screenshot and the conf file? Not gonna put this in issues 'cause it could be something wrong in my config. 
Using arch linux and dosbox staging 0.80.0
![dosboxstaging](https://user-images.githubusercontent.com/964181/209428778-2f47f704-9d5c-4a69-81e4-37edada57c4a.png)

**Comment by rderooy** [comment]
I suspect the issue is that the dosbox-staging window is larger by default than regular dosbox, hence the pixelation is more pronounced. In any case, what you probably want to look at are shaders if you want to smooth things out.

**Comment by GranMinigun** [comment, marked as answer]
Not a bug. The default configuration uses OpenGL output with an oversampling shader, which is something in between nearest-neighbour and bilinear scaling - to keep pixels both as uniform and as sharp as reasonably possible. If you prefer linear output interpolation, then either switch `glshader` to `none` (or perhaps check out other included shaders), or `output` to `texture`. Mind you, CRT monitors didn't blur the image, and a lot of games used display modes with vertical pixels (320x200 has 4:3 aspect ratio, for instance), and that's the reasons behind Staging's default settings.

**Comment by Paolo86cripple** [comment]
Switching to shaders=none worked flawlessly

**Comment by johnnovak** [comment]
> Why is dosbox staging video output so pixelated compared to other versions of Dosbox?

Because out-of-the-box it displays an image much more authentic to how IBM PC video looks on actual CRT monitors compared to other versions of DOSBox 😎 (sorry, couldn't resist!)

The default output of DOSBox Staging is actually very close to how graphics on a VGA monitor looks, especially at 320x200. Pixels on VGA CRT monitors are big chunky rectangles, there's no discernible blur at all. Yes, 320x200 graphics is *supposed* to look blocky!

Now, if you're using a 24" or larger LCD, using DOSBox in fullscreen makes things look worse because the image and thus the pixels become just too large. In that case, I recommended to use the `viewport_resolution` setting to restrict the size of the emulated image to that of a 14-17" CRT monitor people typically used in the DOS era. Try `viewport_resolution = 960x720`, or `1120 x 700` if you want to make the image a little bit bigger. That way the resulting image looks about 80% the same as it would on a CRT VGA monitor (and I'm not talking about "how I remember things"; I have a bunch of fully functional CRT monitors here at home so I can compare their image to the emulated output on my  LCD side-by-side).

Forget about linear interpolation, that's utter garbage... You might be used to it "from other versions of DOSBox", but it's wrong both from the authenticity perspective, and from an aesthetic point of view (like @GranMinigun mentioned, no CRT ever produced such a shitty picture, unless it was completely broken...)
